G1ANT.Addon.Net/API/ImapManager.cs

```diff
@@ -18,10 +18,7 @@ public sealed class ImapManager
         private void ConnectClient(ImapClient client)
         {
             client.Connect(uri);
-            if (credentials != null)
-                client.Authenticate(credentials);
-            if (!client.IsConnected || !client.IsAuthenticated)
-                throw new Exception("Could not connect or authenticate on the server");
+            authenticator?.Authenticate(client);
```

Comment:
Maybe we should throw exception when authenticator is null.

Reply:
it is possible to not pass authenticator to imap.openex/smtp.openex, this is to allow to connect to the server which doesn't need authentication